ARCHITECTURE.md

```diff
@@ -1,15 +1,15 @@
-cddl-gen Architecture
+ZCBOR Architecture
```

Comment:
CBOR is an abbreviation, but ZBOR isn't, right? It does not stand for "Zephyr Consisie Binary Object Representation"? Its just the name of this project. Hence I would argue that we should not use upper case when referencing the name. Instead it should be called `zcbor` everywhere IMO.